Regression: freeimage (unstable-2020-07-04) broken on darwin

**Describe the bug**
`freeimage` on current master is broken, but previous version was working.
Breakage introduced in https://github.com/NixOS/nixpkgs/pull/92708, partial fix in https://github.com/NixOS/nixpkgs/pull/110729.

**To Reproduce**
1. `git checkout 14b5ddf08504fa23b949c3dcf84db17035560fbb`
2. `nix-env --file . --profile /nix/var/nix/profiles/per-user/$USER/testing -iA freeimage`

**Expected behavior**
Successfully install freeimage.

**Actual behavior**
```
make -f Makefile.osx install
make[1]: Entering directory '/private/tmp/nix-build-freeimage-unstable-2020-07-04.drv-0/svn-r1859/FreeImage/trunk'
install Source/FreeImage.h /usr/local/include
install: cannot create regular file '/usr/local/include/FreeImage.h': Permission denied
make[1]: *** [Makefile.osx:107: install] Error 1
```

**Additional context**
See comments from https://github.com/NixOS/nixpkgs/pull/92708#issuecomment-824435951 onwards.

**Notify maintainers**
cc @ricicle @L-as

**Metadata**
 - system: `"x86_64-darwin"`
 - host os: `Darwin 20.3.0, macOS 10.16`
 - multi-user?: `no`
 - sandbox: `no`
 - version: `nix-env (Nix) 2.3.10`
 - channels(uri): `"nixpkgs-21.05pre283153.7919518f023"`
 - nixpkgs: `/Users/uri/.nix-defexpr/channels/nixpkgs`

--- a/pkgs/development/libraries/freeimage/default.nix
+++ b/pkgs/development/libraries/freeimage/default.nix
@@ -29,6 +29,10 @@ stdenv.mkDerivation {
 
   preInstall = ''
     mkdir -p $INCDIR $INSTALLDIR
+  ''
+  # Workaround for Makefiles.osx not using ?=
+  + lib.optionalString stdenv.isDarwin ''
+    makeFlagsArray+=( "INCDIR=$INCDIR" "INSTALLDIR=$INSTALLDIR" )
   '';
 
   postInstall = lib.optionalString (!stdenv.isDarwin) ''

